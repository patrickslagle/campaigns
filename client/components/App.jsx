import React, { Component } from 'react';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userCity: '',
      userIndustry: '',
      userCompanySize: '',
    };
    this.fetchUserData = this.fetchUserData.bind(this);
  }

  componentDidMount() {
    this.fetchUserData();
  }

  fetchUserData() {
    fetch('/userData')
      .then((response) => {
        if (response.ok) return response.json();
        return Promise.reject(new Error('Unable to fetch user data'));
      })
      .then((userData) => {
        const { City, Industry, Size } = userData[0];
        this.setState({
          userCity: City,
          userIndustry: Industry,
          userCompanySize: Size,
        });
      })
      .catch(err => console.error(err));
  }


  render() {
    return (
      <div>
        <div>{this.state.userCity} - {this.state.userIndustry} - {this.state.userCompanySize}</div>
      </div>
    )
  }
}