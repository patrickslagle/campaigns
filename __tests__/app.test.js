import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import App from '../client/components/app.jsx';
import CampaignPhoto from '../client/components/campaign-photo.jsx';

configure({ adapter: new Adapter() });

test('App component renders properly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

test('CampaignPhoto component renders properly without user data', () => {
  const wrapper = shallow(
    <CampaignPhoto
      userCity=""
      userCompanySize=""
      userIndustry=""
      priority={[]}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('CampaignPhoto component renders properly with user data', () => {
  const wrapper = shallow(
    <CampaignPhoto
      userCity="Austin"
      userCompanySize="0-50"
      userIndustry="Software"
      priority={['Austin', 'SanFrancisco', 'Software', 'Sports', '0-50', '100-200']}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('Test that Austin image is rendered correctly', () => {
  const wrapper = shallow(
    <CampaignPhoto
      userCity="Austin"
      userCompanySize="0-50"
      userIndustry="100-200"
      priority={['Austin', 'SanFrancisco', 'Software', 'Sports', '0-50', '100-200']}
    />);
  expect(wrapper.find("img").prop('src')).toBe('./assets/Austin.jpg')
});

test('Test that San Fransisco image is rendered correctly', () => {
  const wrapper = shallow(
    <CampaignPhoto
      userCity="San Francisco"
      userCompanySize="0-50"
      userIndustry="100-200"
      priority={['Austin', 'San Francisco', 'Software', 'Sports', '0-50', '100-200']}
    />);
  expect(wrapper.find("img").prop('src')).toBe('./assets/SanFrancisco.jpg')
});

