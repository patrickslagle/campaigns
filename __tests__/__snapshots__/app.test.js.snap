// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders properly 1`] = `
<div
  id="app"
>
  <CampaignPhoto
    priority={
      Array [
        "Austin",
        "San Francisco",
        "Software",
        "Sports",
        "0-50",
        "100-200",
      ]
    }
    userCity=""
    userCompanySize=""
    userIndustry=""
  />
</div>
`;

exports[`CampaignPhoto component renders properly with user data 1`] = `
<div
  id="campaign-container"
>
  <img
    alt="campaign"
    id="campaign"
    src="./assets/Austin.jpg"
  />
</div>
`;

exports[`CampaignPhoto component renders properly without user data 1`] = `
<div
  id="campaign-container"
>
  <img
    alt="campaign"
    src="./assets/loading.gif"
  />
</div>
`;
